PSReadLine/DynamicHelp.cs

```diff
@@ -0,0 +1,289 @@
+
+            if (!String.IsNullOrEmpty(commandName) && isFullHelp)
+            {
+                if (helpContent is String fullHelp && !String.IsNullOrEmpty(fullHelp))
+                {
+                    string regexPatternToScrollTo = null;
+
+                    if (!String.IsNullOrEmpty(parameterName))
+                    {
+                        string upper = parameterName[0].ToString().ToUpperInvariant();
+                        string lower = parameterName[0].ToString().ToLowerInvariant();
+                        string remainingString = parameterName.Substring(1);
+                        regexPatternToScrollTo = $"-[{upper}|{lower}]{remainingString} [<|\\[]";
+                    }
+
+                    _mockableMethods.WriteToPager(fullHelp, regexPatternToScrollTo);
+                }
+            }
+            else if (!String.IsNullOrEmpty(commandName) && !String.IsNullOrEmpty(parameterName))
+            {
+                if (helpContent is PSObject paramHelp)
+                {
+                    WriteParameterHelp(paramHelp);
+                }
+            }
+        }
+
+        private void DynamicHelpImpl(bool isFullHelp)
+        {
+            _pager ??= new Pager();
+
+            if (InViInsertMode())   // must close out the current edit group before engaging menu completion
+            {
+                ViCommandMode();
+                ViInsertWithAppend();
+            }
+
+            int cursor = _singleton._current;
+            string commandName = null;
+            string parameterName = null;
+
+            foreach(var token in _singleton._tokens)
+            {
+                if (token.TokenFlags == TokenFlags.CommandName)
+                {
+                    commandName = token.Text;
+                }
+
+                var extent = token.Extent;
+                if (extent.StartOffset <= cursor && extent.EndOffset >= cursor)
+                {
+                    if (token.Kind == TokenKind.Parameter)
+                    {
+                        parameterName = ((ParameterToken)token).ParameterName;
+                        break;
+                    }
+                }
+            }
+
+            WriteDynamicHelpContent(commandName, parameterName, isFullHelp);
+        }
+
+        private void WriteDynamicHelpBlock(Collection<string> helpBlock)
+        {
+            var bufferWidth = _console.BufferWidth;
+            var colWidth = Math.Min(helpBlock.Max(s => LengthInBufferCells(s)) + 2, bufferWidth);
+            int columns = 1;
+
+            var dynHelp = new MultilineDisplayBlock
+            {
+                Singleton = this,
+                ColumnWidth = colWidth,
+                Rows = (helpBlock.Count + columns - 1) / columns,
+                ItemsToDisplay = helpBlock
+            };
+
+            dynHelp.SaveCursor();
+            dynHelp.DrawMultilineBlock(dynHelp);
+            dynHelp.RestoreCursor();
```

Comment:
The `SaveCursor` and `RestoreCursor` should be built in `DrawMultilineBlock`.

Reply:
Fixed